develop/events.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Events
+description: A guide for using events provided by the Fabric API.
+authors:
+  - dicedpixels
+  - mkpoli
+  - daomephsta
+  - solidblock
+  - draylar
+  - jamieswhiteshirt
+  - PhoenixVX
+  - Juuxel
+  - YanisBft
+  - liach
+---
+
+<!-- Couldn't find GitHub usernames for: fudge, stormyfabric -->
+
+# Events
+
+Fabric API provides a system that allows mods to react to actions or occurrences, also defined as *events* that occur in the game.
+
+Events are hooks that satisfy common use cases and/or provide enhanced compatibility and performance between mods that hook into the same areas of the code. The use of events often substitutes the use of mixins.
+
+Fabric API provides events for important areas in the Minecraft codebase that multiple modders may be interested in hooking into.
+
+Events are represented by instances of `net.fabricmc.fabric.api.event.Event` which stores and calls *callbacks*. Often there is a single event instance for a callback, which is stored in a static field `EVENT` of the callback interface, but there are other patterns as well. For example, `ClientTickEvents` groups several related events together.
+
+## Callbacks
+
+Callbacks are a piece of code that is passed as an argument to an event. When the event is triggered by the game, the passed piece of code will be executed.
+
+### Callback Interfaces
+
+Each event has a corresponding callback interface, conventionally named `<EventName>Callback`. Callbacks are registered by calling `register()` method on an event instance, with an instance of the callback interface as the argument. 
+
+All event callback interfaces provided by Fabric API can be found in the `net.fabricmc.fabric.api.event` package.
+
+## Listening to Events
+
+### A Simple Example
+
+This example registers an `AttackBlockCallback` to damage the player when they hit blocks that don’t drop an item when hand-mined.
+
+@[code lang=java transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/event/Events.java)
+
+### Adding Items to Existing Loot Tables
+
+Sometimes you may want to add items to loot tables. For example, adding your drops to a vanilla block or entity.
+
+The simplest solution, replacing the loot table file, can break other mods. What if they want to change them as well? We’ll take a look at how you can add items to loot tables without overriding the table.
+
+We’ll be adding eggs to the coal ore loot table.
+
+#### Listening to Loot Table Loading
+
+Fabric API has an event that is fired when loot tables are loaded, `LootTableEvents.MODIFY`. You can register a callback for it in your mod initializer. Let’s also check that the current loot table is the coal ore loot table.
+
+```java
+LootTableEvents.MODIFY.register((resourceManager, lootManager, id, tableBuilder, source) -> {
```

Comment:
Should be backed by real code.

Reply:
My intention (rather the original Wiki's) was show the signature. Of course I can move this to the reference mod, but that would mean there's a random empty event in the code. Personally, I'd rather have the final example, which lined from the reference mod.

Reply:
I should have looked more into the code snippet plugin. It can do a bit more. So disregard my previous comment. All the code is now embedded from the reference mod.